libnestutil/block_vector.h

```diff
@@ -604,14 +609,14 @@ inline bv_iterator< value_type_, ref_, ptr_ >& bv_iterator< value_type_, ref_, p
 {
   // If we are still within the block, we can just decrement the block iterator.
   // If not, we need to switch to the previous block.
-  if ( block_it_ != block_vector_->blockmap_[ block_index_ ].begin() )
+  if ( block_it_ != block_vector_it_->begin() )
   {
     --block_it_;
   }
   else
   {
-    --block_index_;
-    current_block_end_ = block_vector_->blockmap_[ block_index_ ].end();
+    --block_vector_it_;
+    current_block_end_ = block_vector_it_->end();
     block_it_ = current_block_end_ - 1;
   }
```

Comment:
To avoid undefined behavior at the left edge of the blockvector, we need to be careful here. I'd suggest the following:
```c++
else if ( block_vector_it != block_vector_->begin() )
{
  --block_vector_it_;
  current_block_end_ = block_vector_it_->end();
  block_it_ = current_block_end_ - 1;
}
else
{
  // We are before begin(), which is undefined. We mark this by moving the
  // outer iterator to an undefined position that will compare as safely smaller 
  // than begin(). According to C++17 Standard, §27.2.6, decrementing an
  // iterator that is equal to begin() yields undefined behavior.
  --block_vector_it; 
}

Reply:
Done! And I also added an equivalent check in `operator++()`.